### PR TITLE
Client constructor shouldn't mutate options bag

### PR DESCRIPTION
**Pull request by sarangan12**
This PR is to address the issue reported in https://github.com/Azure/autorest.typescript/issues/1039. The issue and proposed solution could be found there. 

@joheredi @deyaaeldeen @jeremymeng Please review and approve the PR.

@ramya-rao-a FYI

### CODE REVIEW

**Review by deyaaeldeen — APPROVED** [review]
Looks good, thanks!

File: src/generators/clientContextFileGenerator.ts [hunk 1]
```diff
@@ -278,14 +280,14 @@ function writeDefaultOptions(
...
-  options.userAgentOptions = {
-    userAgentPrefix: userAgentPrefix
-  };
   
   ${addScopes}
   const optionsWithDefaults = {
     ...defaults,
     ...options,
+    userAgentOptions: {
+      userAgentPrefix: userAgentPrefix
```
Comment (jeremymeng) [inline src/generators/clientContextFileGenerator.ts:289]: nit, fine to discard:
```suggestion
      userAgentPrefix
```

**Review by jeremymeng — APPROVED** [review]
Looks good. Is it possible/worth to have unit test coverage?

File: src/generators/clientContextFileGenerator.ts (hunk 1 above)
Comment (sarangan12) [reply src/generators/clientContextFileGenerator.ts]: Done.

**Pull request merged**